File: API/Helpers/UserParams.cs
using System;

namespace API.Helpers;

//Act like DTO, used for query
public class UserParams : PaginationParams
{

    public string? Gender { get; set; }
    public string? CurrentUsername { get; set; }
    public int MinAge { get; set; } = 18;
    public int MaxAge { get; set; } = 100;
    public string OrderBy { get; set; } = "lastActive";
}

File: client/src/app/_models/userParams.ts
import { User } from './user';

//this is class because want to make use of the constructor
export class UserParams {
  gender: string;
  minAge = 18;
  maxAge = 99;
  pageNumber = 1;
  pageSize = 5;
  orderBy = 'lastActive';

  constructor(user: User | null) {
    this.gender = user?.gender === 'female' ? 'male' : 'female';
  }
}
